### PR TITLE
Add Prometheus Agent documentation

### PR DESCRIPTION
**Pull request by QuentinBisson**
### What does this PR do?

Add first documentation of the prometheus agent for customers

### What does it look like?

(If this is more than a textual change, a screenshot can help in some cases to speed up the review process.)

### Any background context you can provide?

We had to deploy a prometheus agent in clusters to be able to monitor kubernetes core components. This pages describe what it is, why we added it and the architecture.

### What is needed from the reviewers?

An overall help in making this better

### Have you maintained the front matter?

Done

### CODE REVIEW

**Review by QuantumEnigmaa — COMMENTED** [review]
In your diagram you wrote `workload cluster prometheus` for the one in the MC. But isn't the point of using agents to get rid of the prometheus pods for each WC and only having a central one in the MC on which each agent will access using Remote Write ?

**Comment by QuentinBisson** [comment]
So no the plan is to go with mimir in the end and not a single Prometheus because it would use too much memory and CPU.

Also, this doc reflects the current status as I cannot really write either Mimir or LTS as our prometheus is currently not LTS :( Maybe @TheoBrigitte or @hervenicol have opinions :)

**Comment by hervenicol** [comment]
Totally agree with @QuentinBisson:
* we should show current status
* long term goal is a "scalable prometheus-like" like Mimir, but prometheus is still here for a while

**Comment by hervenicol** [comment]
Concerning this documentation, I think it would be useful to have this info:
* data from agent to prometheus goes through an ingress
* metrics are in prometheus-remote-writes and nginx dashboard
* maybe some links to related opsrecipes (like https://intranet.giantswarm.io/docs/support-and-ops/ops-recipes/prometheus-agent-remote-write-failed/)
* we can see web-up with the list of targets on the agent with a port-forward
* agent scrapes servicemonitors that have the label `giantswarm.io/team`
* prometheus keeps scrapes through the "old-style" scraping, ie `giantswarm.io/monitoring` label.

**Comment by QuentinBisson** [comment]
@hervenicol sure but this is external documentation to customers. Do we want to go into that much details especially regarding ops-recipe? I will add information about the dashboards and servicemonitors though

**Comment by hervenicol** [comment]
Maybe not all of it in the public doc, but I think all of this should be documented.
As the opsrecipes are private, sure, not links to opsrecipes.
But for all the rest, would you rather split the info in 2 docs, private / public, or have it all on the public one? Is there anything "secret" in it?

**Comment by QuentinBisson** [comment]
There are no secrets no but I'm afraid a lot of it would be confusing because the basic of our monitoring setup is not documented :D

**Comment by QuentinBisson** [comment]
@hervenicol can you take another look? I think I added all your points :)

**Comment by QuentinBisson** [comment]
We can for sure improve it :)

**Comment by QuentinBisson** [comment]
Also thanks a lot for taking the time to review :)

File: src/content/getting-started/observability/monitoring/prometheus/agent/index.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+As the Prometheus Agent is running in your workload cluster, you can gain access to its UI to inspect the list of currently configured targets it is scraping using the following command `kubectl port-forward -n kube-system prometheus-prometheus-agent-0 9090` then access the url `http://localhost:9090/targets` on your favorite browser.
+
+You should see the following:
+
+![Screenshot of Prometheus Agent User Interface](prometheus-agent-ui.png)
+
+### Access to the metrics
+
+The metrics of the Prometheus Agent can be found on the [grafana instance]({{< relref "/getting-started/observability/grafana/access" >}}) running on your management cluster under the `Prometheus / Remote Write` and `NGINX Ingress controller` dashboards
```
Comment (hervenicol) [inline src/content/getting-started/observability/monitoring/prometheus/agent/index.md:86]: Just checked, "prometheus/remote write" dashboard is private :cry:

**Review by hervenicol — APPROVED** [review]
I love it! Plenty of useful info :grinning: 
As for any doc, there's still room for improvement, but it's already good enough and we can improve over time.

File: src/content/getting-started/observability/monitoring/prometheus/agent/index.md (hunk 1 above)
Comment (QuentinBisson) [reply src/content/getting-started/observability/monitoring/prometheus/agent/index.md:86]: I have a pr on dashboards to change this but I need to test it 😂

**Comment by pipo02mix** [comment]
ye4ah good work, sorry but I arrived late

**Pull request merged**